### PR TITLE
Move the Spring Auto-Configuration doc

### PR DESCRIPTION
**Pull request by jaydeluca**
(Continuation from #3758 after making a mistake rebasing)

Related to https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/10083

Move and adapt the documentation Readme: [main/instrumentation/spring/spring-boot-autoconfigure/README.md](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/spring/spring-boot-autoconfigure/README.md)

--------

after rebasing I am now getting the following textlint errors:

```
  159:23  ✓ error  Incorrect usage of the term: “auto configuration”, use “autoconfiguration” instead    terminology
  172:6   ✓ error  Incorrect usage of the term: “auto-configuration”, use “autoconfiguration” instead    terminology
  238:1   ✓ error  Incorrect usage of the term: “Auto-configures”, use “Autoconfigures” instead          terminology
  251:17  ✓ error  Incorrect usage of the term: “Auto Configuration”, use “autoConfiguration” instead    terminology
  262:21  ✓ error  Incorrect usage of the term: “Auto Configuration”, use “autoConfiguration” instead    terminology
  273:21  ✓ error  Incorrect usage of the term: “Auto Configuration”, use “autoConfiguration” instead    terminology
  275:10  ✓ error  Incorrect usage of the term: “auto-configurations”, use “autoconfigurations” instead  terminology
  278:6   ✓ error  Incorrect usage of the term: “auto-configuration”, use “autoconfiguration” instead    terminology
  486:1   ✓ error  Incorrect usage of the term: “Auto-configuration”, use “Autoconfiguration” instead    terminology
  507:37  ✓ error  Incorrect usage of the term: “auto-configuration”, use “autoconfiguration” instead    terminology
```

"Auto-configuration" is how it is referred to in the [spring boot docs](https://docs.spring.io/spring-boot/docs/2.0.x/reference/html/using-boot-auto-configuration.html) so I am hesitant to just go and update everything to "autoconfiguration" without checking first. @jeanbisutti Do you know how I should proceed?

---

**Preview**: https://deploy-preview-3900--opentelemetry.netlify.app/docs/languages/java/automatic/spring-boot/

### CODE REVIEW

**Comment by trask** [comment]
I think ok to go ahead and change them. We have similar linting in Microsoft docs that always gets me with auto-instrumentation --> autoinstrumentation.

**Comment by jeanbisutti** [comment]
@jaydeluca I see #3903 that seems to fix the spelling issues.

**Comment by chalin** [comment]
> @jaydeluca I see #3903 that seems to fix the spelling issues.

Actually, after #3903 lands, you'll be able to fix the spelling issue automatically using this command:

```console
$ npm run check:text -- --fix
```

(You can run that command already, but you'll need to tweak a few instances of the rewritten words.)

**Review by chalin — COMMENTED** [review]
We're getting closer! See inline for copyedits.

File: content/en/docs/languages/java/automatic/spring-boot.md
Comment (chalin) [inline content/en/docs/languages/java/automatic/spring-boot.md:3]: Change the title to `Spring Boot Starter`, but keep the linkTitle as is.

File: content/en/docs/languages/java/automatic/spring-boot.md
Comment (chalin) [inline content/en/docs/languages/java/automatic/spring-boot.md:34]:
```suggestion
```

File: content/en/docs/languages/java/automatic/spring-boot.md
Comment (chalin) [inline content/en/docs/languages/java/automatic/spring-boot.md:36]: Promote all headings by one level in this page, e.g.:

```suggestion
## Dependency management
```

**Review by chalin — CHANGES_REQUESTED** [review]
Regarding the **organisation / structure of this page**. If you look at the TOC currently, it looks like this:

image

That doesn't make sense to me to have a single top-level heading with pretty much all of the page content under that single heading. As mentioned at the start of this page "this page documents the OpenTelemetry starter" so I think that we should:

- Change the page title to "Spring Boot Starter", to match the actual content of the page
- The heading "OpenTelemetry Spring starter" should be dropped
- All other headings should be promoted one level up.

**Comment by jeanbisutti** [comment]
> Regarding the **organisation / structure of this page**. If you look at the TOC currently, it looks like this:
> 
> image
> That doesn't make sense to me to have a single top-level heading with pretty much all of the page content under that single heading. As mentioned at the start of this page "this page documents the OpenTelemetry starter" so I think that we should:
> 
> * Change the page title to "Spring Boot Starter", to match the actual content of the page
> * The heading "OpenTelemetry Spring starter" should be dropped
> * All other headings should be promoted one level up.

@chalin There is a mistake. Thanks for having pointed this. `Other configurations` should be a top level heading.

It may have another topic. The Spring Boot page should probably only contain this section explaining when to use the Java agent or the starter: 
image

The rest of the page could propably move to another one related to starters and autoconfigurations. Perhaps it could be done in another PR? I could take care of it.

**Comment by chalin** [comment]
> The rest of the page could propably move to another one related to starters and autoconfigurations. Perhaps it could be done in another PR? I could take care of it.

SGTM, and I'm ok with approaching this incrementally. Please create an issue to track the required changes: you can simply create a new issue from https://github.com/open-telemetry/opentelemetry.io/pull/3900#issuecomment-1918850387.

I'll wait for my other comments to be addressed before making a (final) review pass. Thanks!

**Comment by jeanbisutti** [comment]
> Please create an issue to track the required changes: you can simply create a new issue from [#3900 (comment)](https://github.com/open-telemetry/opentelemetry.io/pull/3900#issuecomment-1918850387).

Done: https://github.com/open-telemetry/opentelemetry.io/issues/3911

**Comment by jaydeluca** [comment]
I really appreciate all the help from ya'll 🙏

**Review by jeanbisutti — APPROVED** [review]

**Review by chalin — APPROVED** [review]
❤️ 

So, @jeanbisutti, this actually closes #3911?

File: content/en/docs/languages/java/automatic/spring-boot.md
Comment (chalin) [inline content/en/docs/languages/java/automatic/spring-boot.md:7]:
```suggestion
cSpell:ignore: autoconfigurations autoconfigures datasource logback springboot webflux webmvc
```

**Comment by jeanbisutti** [comment]
> ❤️
> 
> So, @jeanbisutti, this actually closes #3911?

@chalin It does not close #3911. All the non-agent doc is still in the Spring Boot page.

**Comment by jeanbisutti** [comment]
@open-telemetry/docs-approvers FYI, the Spring Boot page documents instrumentation features for Spring Boot and developed in the https://github.com/open-telemetry/opentelemetry-java-instrumentation repository.

**Comment by trask** [comment]
> @open-telemetry/java-approvers PTAL

fyi @jeanbisutti is in @open-telemetry/java-instrumentation-approvers now

**Comment by cartermp** [comment]
Aaaaand we're in!

**Comment by trask** [comment]
thx @jaydeluca ❤️

**Pull request merged**
